SIEM Case Icons

MichaelMarcialis opened the pull request:
### Summary

Added new icons to EUI to support the needs of the upcoming SIEM case workflow integration feature. Including the following icons:

- folderCheck
- folderExclamation
~- logoServicenow~
- push
- quote
- reporter
- users

[Issue](https://github.com/elastic/kibana/issues/58546)

![image](https://user-images.githubusercontent.com/3884767/75373074-8ef5c980-5897-11ea-8f49-a74f1e182ffe.png)

### Checklist

- [x] Check against **all themes** for compatibility in both light and dark modes
~- [ ] Checked in **mobile**~
- [x] Checked in **IE11** and **Firefox**
~- [ ] Props have proper **autodocs**~
- [x] Added **documentation** examples
- [x] Added or updated **jest tests**
~- [ ] Checked for **breaking changes** and labeled appropriately~
~- [ ] Checked for **accessibility** including keyboard-only and screenreader modes~
- [x] A [changelog](https://github.com/elastic/eui/blob/master/CHANGELOG.md) entry exists and is marked appropriately

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_2935/

cchaos commented:
I mentioned in slack that we'd like to not add more third-party logos if possible since EuiIcon can support custom SVG's. 

Also, I've checked these on a non-retina screen and while the new `case_` icons match our folder icons, all of them are blurry in this area on non-retina when I really think they don't need to be.

<img width="473" alt="Screen Shot 2020-02-26 at 11 19 51 AM" src="https://user-images.githubusercontent.com/549577/75365742-9e6f1580-588b-11ea-841f-3f9d008ccdd9.png">

Other than that, the designs of the icons look great. My only other thought goes towards naming.

I'd propose the following:

- `people` -> `users`: We already have a `user` icon and it would be nice to just make this one the plural version.
- `case_open` -> `folderExclamation`: Both `case_` icons are just folders with extra indicators, I'd label them as such, since we try to be explanatory of what the icon **is** and not what it **represents**. That way they're more generalized for universal usage. 
- `case_closed` -> `folderCheck`: Same reason above

MichaelMarcialis commented:
Thanks for the comments, @cchaos!

> I mentioned in slack that we'd like to not add more third-party logos if possible since EuiIcon can support custom SVG's.

Ah, sorry about that. I've removed it and will notify my engineers to use a custom SVG.

> Also, I've checked these on a non-retina screen and while the new case_ icons match our folder icons, all of them are blurry in this area on non-retina when I really think they don't need to be.

Yeah, I noticed that when repurposing the older folder icons. I've updated both the new and old folder icons to sit better on the pixel grid.

> My only other thought goes towards naming.

All updated, per your recommendations. Thanks, again!

cchaos commented on CHANGELOG.md:
```diff
@@ -2,6 +2,7 @@
 
 - Added SASS variables for text variants of the primary palette `$euiColorPrimaryText`, `$euiColorSecondaryText`, etc... Updated components to use these new variables. ([#2873](https://github.com/elastic/eui/pull/2873))
 - Updated SASS mixin `makeHighContrastColor()` to default `$background: $euiPageBackgroundColor` and `$ratio: 4.5`. Created `makeGraphicContrastColor()` for graphic specific contrast levels of 3.0. ([#2873](https://github.com/elastic/eui/pull/2873))
+- Added `caseClosed`, `caseOpen`, `logoServicenow`, `people`, `push`, `quote` and `reporter` icons ([#2935](https://github.com/elastic/eui/pull/2935))
```
You'll need to update your CL now to fix the naming and also add in that you modified the other folder icons.

MichaelMarcialis replied:
Good catch. All updated. Thanks!

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_2935/

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_2935/
